sympy/solvers/solveset.py

```diff
@@ -978,6 +989,349 @@ def _solveset(f, symbol, domain, _check=False):
+    invocation condition inside `Add` case. Once the place for
+    identification helper is determined, you can add a call to solving
+    helper. For your class of equation you need to define your own
+    identification and solving helpers. The identification helper
+    should be implemented for generalised cases and should return
+    either `True` if the given equation belongs to the class otherwise
+    `False`. Solving helpers needs to be implemented with such
+    heuristics or algorithms that solves most of the equations belonging
+    to that class. The value returned from this helper should be handled
+    properly in the main (\_transolve) function itself. It could be either
+    the exact solution or a reduced form of the equation which
+    `solveset` can handle.
+
+    Apart from this, few other things needs to be taken care while adding
+    an equation solver:
+
+    - Naming conventions:
+      Name of the identification helper should be like  `\_check\_class`
+      where `class` will be the name or abbreviation of the class of
+      equation. The solving helper will be named as `\_class\_solver`.
+      For example: for exponential equation it becomes `\_check\_expo`
+      and `\_expo\_solver`.
+    - Be sure to consider consider corner cases.
+    - Adding tests for each helper.
+    - Adding a docstring to your helper that describes the method
+      implemented with its proof of correctness by explaining with few
+      examples. Since most of the methods devised to identify and solve
+      particular class of equations are heuristics therefore it
+      becomes necessary to provide proof of correctness of these methods.
+      Following things needs to be included while writing the
+      documentation for the helpers:
+
+      - What is the purpose of the helper.
+      - How it solves the equation and how does it identifies the equation.
+      - Examples should be included to prove its correctness.
+      - What are the input parameters and what does the helper returns.
+    """
+
+    # invert_complex handles the call to the desired inverter based
+    # on the domain specified.
+    lhs, rhs_s = invert_complex(f, 0, symbol, domain)
+
+    result = ConditionSet(symbol, Eq(f, 0), domain)
+
+    if rhs_s is S.EmptySet:
+        result = S.EmptySet
+    else:
+        rhs = rhs_s.args[0]
+        # do we need a loop for rhs_s?
+        # Can't determine a case as of now.
+        if lhs == symbol:
+            result = rhs_s
+
+        elif lhs.is_Add:
+            # trying to convert to P.O.S form
+            equation = factor(powdenest(lhs - rhs))
+
+            if equation.is_Mul:
+                result = _solveset(equation, symbol, domain)
+
+            # check if it is exponential type equation
+            elif _check_expo(equation, symbol):
+                new_f = _expo_solver(equation)
+                result = _solveset(new_f, symbol, domain)
```

Comment:
what if `new_f` is same as given `f`?

Reply:
I don't think this will happen, because even if an equation gets into the condition, `new_f` will contain expression in `log`. which will be different than `f`.